src/Microsoft.AspNetCore.Sockets.Abstractions/ChannelConnection.cs

```diff
@@ -24,13 +24,12 @@ public class ChannelConnection<T> : Channel<T>, IDisposable
         public Channel<T> Input { get; }
         public Channel<T> Output { get; }
 
-        public override ReadableChannel<T> In => Input;
-
-        public override WritableChannel<T> Out => Output;
-
         public ChannelConnection(Channel<T> input, Channel<T> output)
         {
+            In = input;
```

Comment:
I wonder if we need both - In and Input...

Reply:
We need it so we have access to the actual channel for completing.